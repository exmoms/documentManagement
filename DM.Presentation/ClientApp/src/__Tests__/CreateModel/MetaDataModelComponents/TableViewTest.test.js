import React from "react";
import ReactDom from "react-dom";
import TableView from "../../../components/CreateModel/MetaDataModelComponent/TableView.js";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import TablePagination from "@material-ui/core/TablePagination";
afterEach(cleanup);
describe("Table View", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    const Columns = [
      { id: "AggregateName", label: "Aggregation Name", minWidth: 170 },
      { id: "metaDataModelName", label: "Aggregated Model", minWidth: 100 },
    ];
    const Rows = [
      {
        metaDataModelName: "Meta Data Model test 1",
        AggregateName: " Aggregated Name 1",
      },
      {
        metaDataModelName: "Meta Data Model test 2",
        AggregateName: " Aggregated Name 2",
      },
    ];
    ReactDom.render(<TableView rows={Rows} columns={Columns} />, div);
  });

  test("matches snapshot", () => {
    const Columns = [
      { id: "AggregateName", label: "Aggregation Name", minWidth: 170 },
      { id: "metaDataModelName", label: "Aggregated Model", minWidth: 100 },
    ];
    const Rows = [
      {
        metaDataModelName: "Meta Data Model test 1",
        AggregateName: " Aggregated Name 1",
      },
      {
        metaDataModelName: "Meta Data Model test 2",
        AggregateName: " Aggregated Name 2",
      },
    ];
    const wrapper = shallow(<TableView rows={Rows} columns={Columns} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("TablePagination tests", () => {
    const Columns = [
      { id: "AggregateName", label: "Aggregation Name", minWidth: 170 },
      { id: "metaDataModelName", label: "Aggregated Model", minWidth: 100 },
    ];
    const Rows = [
      {
        metaDataModelName: "Meta Data Model test 1",
        AggregateName: " Aggregated Name 1",
      },
      {
        metaDataModelName: "Meta Data Model test 2",
        AggregateName: " Aggregated Name 2",
      },
    ];
    const wrapper = shallow(<TableView rows={Rows} columns={Columns} />);
    //click onChangePage
    expect(wrapper.find(TablePagination).props().page).toBe(0);
    act(() => {
      wrapper
        .find(TablePagination)
        .props()
        .onChangePage({ target: { value: "" } }, 1);
    });
    expect(wrapper.find(TablePagination).props().page).toBe(1);
    //click onChangeRowsPerPage
    expect(wrapper.find(TablePagination).props().rowsPerPage).toBe(10);
    act(() => {
      wrapper
        .find(TablePagination)
        .props()
        .onChangeRowsPerPage({ target: { value: 5 } });
    });
    expect(wrapper.find(TablePagination).props().rowsPerPage).toBe(5);
    expect(wrapper.find(TablePagination).props().page).toBe(0);
    //check the props of TablePagination
    expect(
      wrapper.find(TablePagination).props().rowsPerPageOptions
    ).toStrictEqual([5, 10, 15]);
    expect(wrapper.find(TablePagination).props().count).toBe(2);
  });
});
