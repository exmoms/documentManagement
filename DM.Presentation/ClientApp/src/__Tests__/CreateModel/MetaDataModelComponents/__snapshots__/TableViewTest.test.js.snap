// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table View matches snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(TableContainer))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Table))
      aria-label="sticky table"
      stickyHeader={true}
    >
      <WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableCell))>
            #
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="AggregateName"
            style={
              Object {
                "minWidth": 170,
              }
            }
          >
            Aggregation Name
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="metaDataModelName"
            style={
              Object {
                "minWidth": 100,
              }
            }
          >
            Aggregated Model
          </WithStyles(ForwardRef(TableCell))>
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableBody))>
        <WithStyles(ForwardRef(TableRow))
          hover={true}
          key="0"
          role="checkbox"
          tabIndex={-1}
        >
          <WithStyles(ForwardRef(TableCell))>
            1
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="AggregateName"
          >
             Aggregated Name 1
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="metaDataModelName"
          >
            Meta Data Model test 1
          </WithStyles(ForwardRef(TableCell))>
        </WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableRow))
          hover={true}
          key="1"
          role="checkbox"
          tabIndex={-1}
        >
          <WithStyles(ForwardRef(TableCell))>
            2
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="AggregateName"
          >
             Aggregated Name 2
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="metaDataModelName"
          >
            Meta Data Model test 2
          </WithStyles(ForwardRef(TableCell))>
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableBody))>
    </WithStyles(ForwardRef(Table))>
  </WithStyles(ForwardRef(TableContainer))>
  <WithStyles(ForwardRef(TablePagination))
    component="div"
    count={2}
    onChangePage={[Function]}
    onChangeRowsPerPage={[Function]}
    page={0}
    rowsPerPage={10}
    rowsPerPageOptions={
      Array [
        5,
        10,
        15,
      ]
    }
  />
</WithStyles(ForwardRef(Paper))>
`;
